Source/astcenc_block_sizes2.cpp

```diff
@@ -659,12 +659,12 @@ static void construct_block_size_descriptor_2d(
 #endif
 
 	// then construct the list of block formats
+	int idx = 0;
```

Comment:
We get some name shadowing errors if compiled with `-Werror=shadow` (some of the code lower down also has an `idx` variable). Can we rename the new variable to `bm_idx`, or something like that; shadowing makes future code review harder.